scripts/build/gen_isr_tables.py

```diff
@@ -14,6 +14,9 @@
 from elftools.elf.sections import SymbolTableSection
 
 ISR_FLAG_DIRECT = (1 << 0)
+ISR_FLAG_SHARE  = (1 << 1)
+
+MAX_SHARED_ISRS = 4
```

Comment:
Is there a specific reason for this particular number? It might be worth putting a justification in the comment, or describe tradeoffs of making it larger or smaller.

Reply:
More nitpicky still: why put a limit in?  Both a singly-linked list or a null-terminated array can be generated at build time by this script (and the list variant could be then modified at runtime by e.g. irq_connect_dynamic()).

Reply:
I've removed the max number of shared ISRs. Any number of ISRs can been shared.